src/DynamicQuantities.jl

```diff
@@ -1,14 +1,6 @@
 module DynamicQuantities
 
-import TOML: parsefile
-
-const PACKAGE_VERSION = try
-    let project = parsefile(joinpath(pkgdir(@__MODULE__), "Project.toml"))
-        VersionNumber(project["version"])
-    end
-catch
-    VersionNumber(0, 0, 0)
-end
+using PackageExtensionCompat
```

Comment:
```suggestion
import PackageExtensionCompat: @require_extensions
```

(Personal preference to use explicit imports so its easier to figure out what symbol came from where, and to avoid namespace collisions)

Reply:
Sure, I like that as well. I'd suggest
```suggestion
using PackageExtensionCompat: @require_extensions
```
though since the macro is only used but not extended in the package.

Reply:
> though since the macro is only used but not extended in the package.

Oh, I never knew this. Is this a standard strategy (`using` vs `import`) for indicating intended usage of a symbol?

I've been using BlueStyle for most of my packages which enforces the usage of `import` everywhere, which is why I stick to it in my packages (including internal imports; see `symbolic_dimensions.jl`). So I guess even if it doesn't match the semantic meaning I'd prefer to be consistently against style guidelines rather than inconsistently correct. (If that's okay with you?). But maybe I'll just switch everything over some time...

Reply:
Oh, weird. Not sure why I thought it did.

> So `using` is a bit safer IMO since it prevents you from unintentionally extending functions.

This is a **really** good point! Okay maybe we can try to switch them all (at once) in another PR. (And now I need to also do that in all my other libraries...)

Reply:
Reference: https://docs.julialang.org/en/v1/manual/modules/#using-and-import-with-specific-identifiers,-and-adding-methods

As an additional comment, if you would like to only load the module itself you can use `using MyPackage: MyPackage`.

BlueStyle actually explicitly notes that `using` should be prefered over `import`, even when extending functions: https://github.com/invenia/BlueStyle#module-imports